src/app/js/public/Drawer.js

```diff
@@ -22,126 +21,71 @@ const styles = StyleSheet.create({
     drawer: {
+        zIndex: 1001,
         position: 'relative',
         height: '100vh',
-        transition: `transform ${ANIMATION_DURATION}ms`,
         transitionTimingFunction: 'ease-in-out',
         backgroundColor: '#f8f8f8',
         width: DRAWER_WIDTH,
-        overflowX: 'scroll',
-        boxShadow: '0 2px 1rem #777',
+        overflowY: 'auto',
     },
     closed: {
         transform: `translateX(-${DRAWER_WIDTH}px)`,
     },
+    closing: {
+        transform: `translateX(-${DRAWER_WIDTH}px)`,
+        boxShadow: '0 2px 1rem #777',
+    },
     open: {
         transform: 'translate(0)',
-    },
-    openButton: {
-        position: 'absolute',
-        top: 'calc(50vh - 50px)',
-        height: 100,
-        width: 100,
-        left: -80,
-        backgroundColor: 'lightgray',
-        borderRadius: '50%',
-        cursor: 'pointer',
-        pointerEvents: 'auto',
-        transition: 'transform 100ms',
-        transitionTimingFunction: 'ease-in-out',
-        ':hover': {
-            transform: 'translateX(5px)',
-            filter: 'brightness(0.8)',
-        },
-        ':after': {
-            content: '"🔍"',
-            position: 'absolute',
-            top: 'calc(50% - 10px)',
-            right: 5,
-        },
+        boxShadow: '0 2px 1rem #777',
     },
     mask: {
         position: 'absolute',
+        zIndex: 1000,
         top: 0,
-        left: DRAWER_WIDTH,
+        left: 0,
         height: '100vh',
-        width: `calc(100vw - ${DRAWER_WIDTH}px)`,
+        width: `100vw`,
     },
 });
 
-class Drawer extends Component {
-    state = {
-        open: false,
-        showChildren: false,
-    };
-
-    openDrawer = () => {
-        const { open } = this.state;
-
-        if (open) {
-            return;
-        }
-
-        this.setState({ open: true, showChildren: true });
-    };
-
-    closeDrawer = () => {
-        const { open } = this.state;
-
-        if (!open) {
-            return;
-        }
-
-        this.setState({ open: false }, () =>
-            setTimeout(() => {
-                this.setState({ showChildren: false });
-            }, ANIMATION_DURATION),
-        );
-    };
-
-    renderClosed = () => (
-        <div className={css(styles.openButton)} onClick={this.openDrawer} />
-    );
-
-    render() {
-        const { open, showChildren } = this.state;
-        const { children } = this.props;
-
-        return (
+const Drawer = ({ children, status, animationDuration, onClose }) => (
+    <div
+        className={classnames(
+            'drawer-container',
+            css(styles.container),
+            css(
+                styles[status === 'open' ? 'openContainer' : 'closedContainer'],
+            ),
+        )}
+    >
+        <div
+            className={classnames(
+                'drawer',
+                css(styles.drawer),
+                css(styles[status]),
+            )}
+            style={{
+                transition: `transform ${animationDuration}ms`,
```

Comment:
Why this style isn't in the `drawer` class?

Reply:
to have acces to the animationDuration props